### ISSUE TITLE
iOS app crashing after login

### ISSUE THREAD

**Issue by omdv** [bitwarden/mobile#363]
Hello. I am using the iOS app with the self-hosted server. I confirmed that the server works with desktop app. It also works with iOS app for creating accounts. If I attempt to login it just crashes.

The relevant portion of iOS logs is below.

```
{"app_name":"BitwardeniOS","timestamp":"2018-08-29 21:23:30.26 -0500","app_version":"1.18.1","slice_uuid":"fb345c5f-95a8-3b1d-b2a5-820841939644","adam_id":1137397744,"build_version":"36","bundleID":"com.8bit.bitwarden","share_with_app_devs":false,"is_first_party":false,"bug_type":"109","os_version":"iPhone OS 11.4.1 (15G77)","incident_id":"28C07BBA-6791-408D-88F9-2B58CE885A78","name":"BitwardeniOS"}
Incident Identifier: 28C07BBA-6791-408D-88F9-2B58CE885A78
CrashReporter Key:   67246e7074117d9e876d3dd2d82e03c2de2b28d6
Hardware Model:      iPhone9,3
Process:             BitwardeniOS [15681]
Path:                /private/var/containers/Bundle/Application/4733AB34-9D77-4664-9FE6-3E66C2648D8B/BitwardeniOS.app/BitwardeniOS
Identifier:          com.8bit.bitwarden
Version:             36 (1.18.1)
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           com.8bit.bitwarden [1247]


Date/Time:           2018-08-29 21:23:29.9815 -0500
Launch Time:         2018-08-29 20:57:53.6800 -0500
OS Version:          iPhone OS 11.4.1 (15G77)
Baseband Version:    2.03.12
Report Version:      104

Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Triggered by Thread:  0

Application Specific Information:
abort() called

Filtered syslog:
None found

Last Exception Backtrace:
(0x183162d8c 0x18231c5ec 0x1831629f0 0x101e9b968 0x101e90a48 0x101e9081c 0x101e91b78 0x100dc2888 0x1022fd304 0x1023853b8 0x10238a334 0x102389eb0 0x102389d04 0x1022ec374 0x1022eb80c 0x1022e187c 0x100df7238 0x100cae14c 0x100cb4524 0x10171252c 0x100dc2888 0x1022fd304 0x1023839c8 0x1023872ec 0x100aa7bc8 0x100aa8cb8 0x183bae0ec 0x18310b404 0x18310ac2c 0x18310879c 0x183028da8 0x18500e020 0x18d048758 0x101754ce4 0x1016d1f6c 0x1016d1f2c 0x100ad4478 0x100dc2888 0x1022fd304 0x1023839c8 0x102389900 0x1022dd8d4 0x10200c4f4 0x100ad3e8c 0x182ab9fc0)

Thread 0 name:  tid_303  Dispatch queue: com.apple.main-thread
Thread 0 Crashed:
0   libsystem_kernel.dylib        	0x0000000182be92e0 0x182bc7000 + 140000
1   libsystem_pthread.dylib       	0x0000000182d8a288 0x182d87000 + 12936
2   libsystem_c.dylib             	0x0000000182b57d0c 0x182af5000 + 404748
3   BitwardeniOS                  	0x0000000101fe4ef0 0x100a80000 + 22433520
4   CoreFoundation                	0x00000001831631c8 0x18301d000 + 1335752
5   libobjc.A.dylib               	0x000000018231c8c8 0x182314000 + 35016
6   BitwardeniOS                  	0x0000000101fccb3c 0x100a80000 + 22334268
7   libc++abi.dylib               	0x000000018230d37c 0x1822f2000 + 111484
8   libc++abi.dylib               	0x000000018230cccc 0x1822f2000 + 109772
9   libobjc.A.dylib               	0x000000018231c720 0x182314000 + 34592
10  CoreFoundation                	0x00000001831629f0 0x18301d000 + 1333744
11  BitwardeniOS                  	0x0000000101e9b968 0x100a80000 + 21084520
12  BitwardeniOS                  	0x0000000101e90a48 0x100a80000 + 21039688
13  BitwardeniOS                  	0x0000000101e9081c 0x100a80000 + 21039132
14  BitwardeniOS                  	0x0000000101e91b78 0x100a80000 + 21044088
15  BitwardeniOS                  	0x0000000100dc2888 0x100a80000 + 3418248
16  Mono                          	0x00000001022fd304 0x10229c000 + 398084
17  Mono                          	0x00000001023853b8 0x10229c000 + 955320
18  Mono                          	0x000000010238a334 0x10229c000 + 975668
19  Mono                          	0x0000000102389eb0 0x10229c000 + 974512
20  Mono                          	0x0000000102389d04 0x10229c000 + 974084
21  Mono                          	0x00000001022ec374 0x10229c000 + 328564
22  Mono                          	0x00000001022eb80c 0x10229c000 + 325644
23  Mono                          	0x00000001022e187c 0x10229c000 + 284796
24  BitwardeniOS                  	0x0000000100df7238 0x100a80000 + 3633720
25  BitwardeniOS                  	0x0000000100cae14c 0x100a80000 + 2285900
26  BitwardeniOS                  	0x0000000100cb4524 0x100a80000 + 2311460
27  BitwardeniOS                  	0x000000010171252c 0x100a80000 + 13182252
28  BitwardeniOS                  	0x0000000100dc2888 0x100a80000 + 3418248
29  Mono                          	0x00000001022fd304 0x10229c000 + 398084
30  Mono                          	0x00000001023839c8 0x10229c000 + 948680
31  Mono                          	0x00000001023872ec 0x10229c000 + 963308
32  BitwardeniOS                  	0x0000000100aa7bc8 0x100a80000 + 162760
33  BitwardeniOS                  	0x0000000100aa8cb8 0x100a80000 + 167096
34  Foundation                    	0x0000000183bae0ec 0x183a95000 + 1151212
35  CoreFoundation                	0x000000018310b404 0x18301d000 + 975876
36  CoreFoundation                	0x000000018310ac2c 0x18301d000 + 973868
37  CoreFoundation                	0x000000018310879c 0x18301d000 + 964508
38  CoreFoundation                	0x0000000183028da8 0x18301d000 + 48552
39  GraphicsServices              	0x000000018500e020 0x185003000 + 45088
40  UIKit                         	0x000000018d048758 0x18cd2b000 + 3266392
41  BitwardeniOS                  	0x0000000101754ce4 0x100a80000 + 13454564
42  BitwardeniOS                  	0x00000001016d1f6c 0x100a80000 + 12918636
43  BitwardeniOS                  	0x00000001016d1f2c 0x100a80000 + 12918572
44  BitwardeniOS                  	0x0000000100ad4478 0x100a80000 + 345208
45  BitwardeniOS                  	0x0000000100dc2888 0x100a80000 + 3418248
46  Mono                          	0x00000001022fd304 0x10229c000 + 398084
47  Mono                          	0x00000001023839c8 0x10229c000 + 948680
48  Mono                          	0x0000000102389900 0x10229c000 + 973056
49  Mono                          	0x00000001022dd8d4 0x10229c000 + 268500
50  BitwardeniOS                  	0x000000010200c4f4 0x100a80000 + 22594804
51  BitwardeniOS                  	0x0000000100ad3e8c 0x100a80000 + 343692
52  libdyld.dylib                 	0x0000000182ab9fc0 0x182ab9000 + 4032
```

The url of the server is: https://1me53pj8la.execute-api.us-west-2.amazonaws.com/prod
Feel free to create new accounts or try anything with it.

I tried building app from source to debug, but Visual Studio on Mac when debugging with iPhone simulator gives me errors during app initialization and does not reach the login stage. I need to try on Windows. Will appreciate any suggestions.

**Comment by kspearrin** [comment]
Unfortunately the crash log doesn't have much helpful information. Are you able to log into the standard cloud server without issue?

**Comment by omdv** [comment]
Yes, the standard cloud server works OK. I guess this means that the issue is likely with the self-hosted solution.

**Comment by kspearrin** [comment]
Could you send me the timestamp of a crash so that I can see if it is in our production crash logs? Make sure you open the app back up after the crash so that it can be transmitted,

**Comment by omdv** [comment]
Kyle, I thought the log above contains the timestamp. Are you looking for anything else?

**Comment by kspearrin** [comment]
I don't seem to be able to find an associated crash log. I could just try connecting to your server, but I get a "Missing Authentication Token" when trying to access https://1me53pj8la.execute-api.us-west-2.amazonaws.com/prod

**Comment by omdv** [comment]
Thanks for your time, Kyle. I am quite sure this is the issue on the server side, as I am using the serverless implementation hosted on AWS, so I'll close it.

**Comment by corelgott** [comment]
I've also got this issue. I'm running a self hosted bit warden server on a docker host. I've various ios clients connected. Two iPads, two iPhones, several android etc. I've just moved from my iPhone 6 to an iPhone 8. But as soon as I log into my account via the Ios App it crashes. This only happends whenever the password ist correct and the app tries to load the list of logins. When entering a wrong password, the app correctly denies the login.
- I've deleted the app serveral times an reloaded it.
- I've checked the server for updates, but its running the most current version
- several other client (web, ios, android) work fine
- the settings (server, user & pwd) are correct; Checked them over and over.
Any ideas on how to approach this issue? I'm happy to provide debug info if you point me into the right direction.
